render/renderer.go

```diff
@@ -137,10 +139,42 @@ func (r *Renderer) getTileImage(tile *tiled.LayerTile) (image.Image, error) {
 	return r.engine.RotateTileImage(tile, timg), nil
 }
 
+func (r *Renderer) _renderTile(layer *tiled.Layer, i int, x int, y int) error {
+	if layer.Tiles[i].IsNil() {
+		return nil
+	}
+
+	img, err := r.getTileImage(layer.Tiles[i])
+	if err != nil {
+		return err
+	}
+
+	pos := r.engine.GetTilePosition(x, y)
+
+	if layer.Opacity < 1 {
+		mask := image.NewUniform(color.Alpha{uint8(layer.Opacity * 255)})
+
+		draw.DrawMask(r.Result, pos, img, img.Bounds().Min, mask, mask.Bounds().Min, draw.Over)
+	} else {
+		draw.Draw(r.Result, pos, img, img.Bounds().Min, draw.Over)
+	}
+
+	return nil
+}
+
 func (r *Renderer) _renderLayer(layer *tiled.Layer) error {
 
 	var xs, xe, xi, ys, ye, yi int
-	if r.m.RenderOrder == "" || r.m.RenderOrder == "right-down" {
+	var odd bool
```

Comment:
Maybe you can introduce some strategy pattern here already? these if's will become hard to maintain